presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java

```diff
@@ -130,9 +143,47 @@ public PartitionStatistics getTableStatistics(String databaseName, String tableN
     @Override
     public Map<String, PartitionStatistics> getPartitionStatistics(String databaseName, String tableName, Set<String> partitionNames)
     {
-        // TODO implement partition statistics
-        // currently returns a map of partitionName to empty statistics to satisfy presto requirements
-        return partitionNames.stream().collect(toImmutableMap(identity(), (p) -> PartitionStatistics.empty()));
+        Table table = getTable(databaseName, tableName).orElseThrow(() -> new TableNotFoundException(new SchemaTableName(databaseName, tableName)));
+        List<String> dataColumns = table.getDataColumns().stream()
+                .map(Column::getName)
+                .collect(toImmutableList());
+        List<String> partitionColumns = table.getPartitionColumns().stream()
```

Comment:
No need to have this. Use `makePartName` from `MetastoreUtil` instead of `FileUtils.makePartName`